sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -116,7 +116,11 @@ private[sql] object CatalogV2Util {
         newProperties.put(set.property, set.value)
 
       case unset: RemoveProperty =>
-        newProperties.remove(unset.property)
+        val prop = unset.property
+        if (!unset.ifExists && !properties.containsKey(prop) && prop != TableCatalog.PROP_COMMENT) {
```

Comment:
This is from:
https://github.com/apache/spark/blob/e614f34c7a538b1f2c59616689eaea95af85fd54/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L307-L311